src/conf.c

```diff
@@ -2402,3 +2402,77 @@ static int conf__parse_string(char **token, const char *name, char **value, char
 	}
 	return MOSQ_ERR_SUCCESS;
 }
+
+int config__write(struct mosquitto__config *config, const char *file_path) {
+    FILE *file = fopen(file_path, "r");
+    if(file == NULL) {
+        mosquitto_log_printf(MOSQ_LOG_ERR, "Could not open config file!");
+        return 1;
+    }
+
+    FILE *temp_file = fopen("temp_config.conf", "w");
```

Comment:
"/tmp/mqbc/temp_config.conf"

Reply:
I used tmpfile() function to create temporary function, it creates a temporary function in root without a name and deletes automatically either when the file is closed or program terminates.